Makefile

```diff
@@ -10,6 +10,9 @@ ifndef BUILD_TOOLS_DOCKER_IMAGE
 	BUILD_TOOLS_DOCKER_IMAGE := ${BUILD_TOOLS_DOCKER_REPO}:${BUILD_TOOLS_VERSION}
 endif
 
+USER_UID := $(shell id -u)
```

Comment:
Wouldn't it be a better idea to create a new unprivileged user inside the image? `$(shell id -u)` won't change anything if the current user if privileged and that will most likely always be the case.

Reply:
Files created need to be owned by current user.. that's the only reason..

Reply:
ok but do we actually create files other than the snyk reports ?

Reply:
as mentioned, how about creating a user and home directory right inside the Dockerfile instead?

Reply:
How do you know the calling user id? You need to pass it.. ;)